nifi-provenance-reporting-tasks/src/main/java/io/egm/nifi/reporting/EmailProvenanceReporter.java

```diff
@@ -295,21 +330,33 @@ public void indexEvent(final Map<String, Object> event, final ReportingContext c
             getLogger().error("Error sending error email: " + e.getMessage(), e);
         }
     }
-
     public void sendErrorEmail(Map<String, Object> event, ReportingContext context) throws MessagingException {
 
-        String emailSubject = "Error occurred in processor " + event.get("component_name") + " "
-                + "in process group " + event.get("process_group_name");
+        String emailSubject = "[" + context.getProperty(EMAIL_SUBJECT_PREFIX).getValue() + "] : "
+                + "Error occurred in processor " + event.get("component_name") + " "
+                + "in process group " + event.get("process_group_name") +   "from nifi instance: " ;
 
         final Properties properties = new Properties();
         final Session mailSession = this.createMailSession(properties, context);
         final Message message = new MimeMessage(mailSession);
+        InternetAddress[] inetAddressesArray;
+        String specificRecipientAttributeValue = getSpecificRecipientValue(context, event);
+
+        if (context.getProperty(SPECIFIC_RECIPIENT_ATTRIBUTE_NAME).getValue() != null
+                && specificRecipientAttributeValue != null) {
+            inetAddressesArray = (InternetAddress[]) Stream.of(
+                    toInetAddresses(context, specificRecipientAttributeValue, TO),
+                    toInetAddresses(context, specificRecipientAttributeValue, SPECIFIC_RECIPIENT_ATTRIBUTE_NAME)
```

Comment:
It looks weird to call `toInetAddresses` twice with `specificRecipientAttributeValue`. When I look at the code of `toInetAddresses`, I think it will be clearer to have two different methods (the original one and a new one that will just take `specificRecipientAttributeValue` as input parameter)